ocflib/lab/stats.py

```
@@ -139,6 +139,36 @@ def list_desktops(public_only=False):
         c.search(OCF_LDAP_HOSTS, filter, attributes=['cn'])
         return [entry['attributes']['cn'][0] for entry in c.response]
 
+def build_query(host):
+    def add_ocf(hostname):
+        if not hostname.endswith('.ocf.berkeley.edu'):
+            return hostname + '.ocf.berkeley.edu'
+        return hostname
+    query = 'SELECT * FROM `session`'
+    query_args = []
+
+    query_conditions = []
+    query_conditions.append('`host` = %s')
+    query_args.append(add_ocf(host))
+    
+    query += ' WHERE ' + ' AND '.join(query_conditions)
+    query += ' ORDER BY `start` DESC LIMIT 1'
+    # we can't have another user start before current user ends so here we order by start
+
+    return (query, query_args)
+
+def last_used(desktop_host):
+    try:
+        with open('/etc/ocfstats-ro.passwd', 'r') as fin:
+            password = fin.read().strip()
+    except FileNotFoundError:
```

No need to catch this error, I'd just let the function fail.